### PR TITLE
Docs: small typos and grammar fixes

### PR DESCRIPTION
**Pull request by ryanio**
## Description

This PR fixes some typos, grammar, and broken links in the docs.

## Type of change

- [x] Docs typos/grammar

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build-all` and tested the resulting files from `dist` folder in a browser.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have tested my code on the live network.

### CODE REVIEW

File: docs/callbacks-promises-events.rst [hunk 1]
```diff
@@ -4,8 +4,7 @@
 Callbacks Promises Events
 =========================
 
-To help web3 integrate into all kind of projects with different standards
-we provide multiple ways to act on asynchronous functions.
+To help web3 integrate into all kind of projects with different standards we provide multiple ways to act on asynchronous functions.
```
Comment (cgewecke) [inline docs/callbacks-promises-events.rst]:
```suggestion
To help web3 integrate into all kinds of projects with different standards we provide multiple ways to act on asynchronous functions.
```

File: docs/getting-started.rst [hunk 2]
```diff
@@ -26,7 +26,9 @@ First you need to get web3.js into your project. This can be done using the foll
 - pure js: link the ``dist/web3.min.js``
 
 After that you need to create a web3 instance and set a provider.
-Ethereum supported Browsers like Mist or MetaMask will have a ``ethereumProvider`` or ``web3.currentProvider`` available. For  web3.js, check ``Web3.givenProvider``.
+
+Most Ethereum-supported browsers like MetaMask and TrustWallet have an [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) compliant provider available at ``window.ethereum``. For web3.js, check ``Web3.givenProvider``.
+
```
Comment (cgewecke) [inline docs/getting-started.rst:31]: Great 👍

File: docs/include_package-core.rst [hunk 3]
```diff
@@ -289,7 +289,7 @@ extend
 
 Allows extending the web3 modules.
 
-.. note:: You also have ``*.extend.formatters`` as additional formatter functions to be used for in and output formatting. Please see the `source file <https://github.com/ethereum/web3.js/blob/master/packages/web3-core-helpers/src/formatters.js>`_ for function details.
+.. note:: You also have ``*.extend.formatters`` as additional formatter functions to be used for in and output formatting. Please see the `source file <https://github.com/ethereum/web3.js/blob/1.x/packages/web3-core-helpers/src/formatters.js>`_ for function details.
```
Comment (cgewecke) [inline docs/include_package-core.rst]:
```suggestion
.. note:: You also have ``*.extend.formatters`` as additional formatter functions to be used for input and output formatting. Please see the `source file <https://github.com/ethereum/web3.js/blob/1.x/packages/web3-core-helpers/src/formatters.js>`_ for function details.
```

File: docs/web3-eth-contract.rst [hunk 4]
```diff
@@ -372,9 +370,9 @@ handleRevert
     web3.eth.Contract.handleRevert
     contract.handleRevert // on contract instance
 
-The ``handleRevert`` options property does default to ``false`` and will return the revert reason string if enabled on :ref:`send <contract-send>` or :ref:`call <contract-call>` of a contract method.
+The ``handleRevert`` options property defaults to ``false`` and returns the revert reason string if enabled on :ref:`send <contract-send>` or :ref:`call <contract-call>` of a contract method.
 
-.. note:: The revert reason string and the signature does exist as property on the returned error.
+.. note:: The revert reason string and signature exists as property on the returned error.
```
Comment (cgewecke) [inline docs/web3-eth-contract.rst]:
```suggestion
.. note:: The revert reason string and signature are properties on the returned error.
```

File: docs/web3-eth.rst [hunk 5]
```diff
@@ -1828,11 +1824,9 @@ requestAccounts
 
     web3.eth.requestAccounts([callback])
 
-This method will request/enable the accounts from the current environment it is running (Metamask, Status or Mist).
-It doesn't work if you're connected to a node with a default Web3.js provider. (WebsocketProvider, HttpProvidder and IpcProvider)
-This method will only work if you're using the injected provider from a application like Status, Mist or Metamask.
+This method will request/enable the accounts from the current environment. This method will only work if you're using the injected provider from a application like Metamask, Status or TrustWallet. It doesn't work if you're connected to a node with a default Web3.js provider (WebsocketProvider, HttpProvidder and IpcProvider).
 
-For further information about the behavior of this method please read the EIP of it: `EIP-1102 <https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1102.md>`_
+For more information about the behavior of this method please read `EIP-1102: Opt-in account exposure <https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1102.md>`_.
```
Comment (cgewecke) [inline docs/web3-eth.rst:1829]: 👍

**Review by cgewecke — APPROVED** [review]
This is extraordinary!! 💯  💐  🥇 

Thanks so much. Dramatically improves legibility.  Wonderful!

(Saw a couple tiny stragglers as I went through per comments but am astonished by how much you got cleaned up in this sweep.)

**Comment by holgerd77** [comment]
Yeah, really innocent PR title for this impressive walkthrough! 😄 👍

**Comment by ryanio** [comment]
thanks guys!

**Pull request merged**